cinder/volume/drivers/vmware/datastore.py

```diff
@@ -247,6 +252,7 @@ def _get_datastores(self):
 
         return datastores
 
+    @cached(cache=TTLCache(maxsize=128, ttl=300))
```

Comment:
This doesn't only select a DS, but also a host. That host might be down or the datastore might be inaccessible. Not sure if that's fine for 5min when we could prevent it. We also return a datastore summary here - is that used to compute the free space in some way or form? Because then we cannot cache it or risk overprovisioning.